src/api/ApiClient.ts

```diff
@@ -0,0 +1,77 @@
+import Axios from 'axios';
+import type { AxiosInstance } from 'axios';
+
+export type HttpHeaders = {
+  [key: string]: string;
+};
+
+export type RequestConfig = {
+  headers: HttpHeaders;
+};
+
+export class ApiConfiguration {
+  accessToken?: string;
+}
+
+export interface IApiClient {
+  post<Request, Response>(path: string, object: Request, config?: RequestConfig): Promise<Response>;
+  patch<Request, Response>(path: string, object: Request): Promise<Response>;
+  put<Request, Response>(path: string, object: Request): Promise<Response>;
+  get<Response>(path: string): Promise<Response>;
+}
+
+export default class ApiClient implements IApiClient {
+  private client: AxiosInstance;
+
+  protected createAxiosClient(): AxiosInstance {
+    return Axios.create({
+      baseURL: 'http://localhost:3000',
+      responseType: 'json' as const,
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      timeout: 10 * 1000,
+    });
+  }
+
+  constructor() {
+    this.client = this.createAxiosClient();
+  }
+
+  async post<Request, Response>(path: string, body: Request, config?: RequestConfig | undefined): Promise<Response> {
+    try {
+      const response = config
+        ? await this.client.post<Response>(path, body, config)
+        : await this.client.post<Response>(path, body);
+      return response.data;
+    } catch (error) {
+      throw new Error('Something went wrong');
+    }
+  }
+  async patch<Request, Response>(path: string, body: Request): Promise<Response> {
+    try {
+      const response = await this.client.patch<Response>(path, body);
+      return response.data;
+    } catch (error) {
+      throw new Error('Something went wrong');
+    }
+  }
+
+  async put<Request, Response>(path: string, body: Request): Promise<Response> {
+    try {
+      const response = await this.client.put<Response>(path, body);
+      return response.data;
+    } catch (error) {
+      throw new Error('Something went wrong');
+    }
+  }
+
+  async get<Response>(path: string): Promise<Response> {
+    try {
+      const response = await this.client.get<Response>(path);
+      return response.data;
+    } catch (error) {
+      throw new Error('Something went wrong');
+    }
+  }
+}
```

Comment:
Can you think of a way to test this file?

Reply:
Yes created a mock with Axios in the test file and tested whether or not a function was called. I needed to refactor the class to inject the client this file in stead of creating a new instance when I use this class.